systems/plants/RigidBody.cpp

```diff
@@ -62,7 +64,7 @@ void RigidBody::setN(int nq, int nv) {
 }
 
 
-void RigidBody::setJoint(std::unique_ptr<DrakeJoint> new_joint)
+void RigidBody::setJoint(std::unique_ptr<DrakeJoint>& new_joint)
```

Comment:
This was a change with @tkoolen on my branch.  That's a useful article.  I'll switch it back and do the move explicitly -- which was the problem we were trying to avoid.

On Mar 15, 2015, at 2:50 PM, Andres Valenzuela <notifications@github.com> wrote:

> In systems/plants/RigidBody.cpp:
> 
> > @@ -62,7 +64,7 @@ void RigidBody::setN(int nq, int nv) {
> >  }
> >  
> >  
> > -void RigidBody::setJoint(std::unique_ptr<DrakeJoint> new_joint)
> > +void RigidBody::setJoint(std::unique_ptr<DrakeJoint>& new_joint)
> I believe the conventional wisdom is to take unique_ptr arguments by value if you're going to move from them. See 3.c in this article of Herb Sutter's.
> 
> —
> Reply to this email directly or view it on GitHub.
>